bioimageio/spec/commands.py

```diff
@@ -19,9 +22,26 @@ def validate(
     if update_format_inner is None:
         update_format_inner = update_format
 
-    source_name = str(rdf_source.get("name") if isinstance(rdf_source, dict) else rdf_source)
+    if isinstance(rdf_source, (BytesIO, StringIO)):
```

Comment:
I don't fully understand what's going on here, but in general this code would be easier to understand if it would look like
```python
if isinstance(rdf_source, (BytesIO, StringIO)):
   ...
elif isinstance(rdf_source, os.PathLike)
   ...
elif isinstance(rdf_source, str) and rdf_source.startswith("http):
   ...
elif isinstance(rdf_source, str) and re.fullmatch(DOI_REGEX, rdf_source):   
   ...
# some more options I overlooked?   
else:
   if not isinstance(rdf_source): raise ValueError   
```

Reply:
👍 yeah.. that grew a bit over time tbh...